hapi/base.py

```diff
@@ -132,12 +138,18 @@ def _execute_request(self, conn, request):
         return result.body
 
     def _digest_result(self, data):
-        if data and isinstance(data, basestring):
+        if data and isinstance(data, six.string_types):
+            # This is for python 2 version, data is str
             try:
                 data = json.loads(data)
             except ValueError:
                 pass
-
+        else:
+            # This is for python 3 version, data is bytes string and get Gzipped
+            try:
+                data = json.loads(gzip.GzipFile(fileobj=io.BytesIO(data)).read())
```

Comment:
Why is this a GZip file only in Python 3?

Reply:
First of all, `http.client` is a package for python2/3 compatibility. In python2, it will actually still import `httplib` , but in python 3, it will import `http.client`.

Here that variable `data` comes from line 217 `self._digest_result(result.body)` in this file. 

In python 2, `result` is a `httplib.HTTPResponse` instance, in which the body is just a `str` instance.

But in python 3, `result` is a `http.client.HTTPResponse` instance, in which the body is a `bytes` instance. Meanwhile, `result` have one more key-value pair: `'headers': <http.client.HTTPMessage object at 0x110a9e510>` which does not exist in python 2. And the `Content-Encoding` header is `gzip` here.

Reply:
Why is the result gzipped in python 3 and not in python 2? Shouldn't the response header be checked?

Reply:
As I mentioned before, there is no such header information from ```httplib.HTTPResponse``` in python2. The ```data``` here is  just a ```str``` instance. In previous codes, the response header is not checked and ```data``` is passed to ```json.loads(data)``` directly.

Reply:
How does this code work? Either the data on the wire is gzipped or it isn't -- this is determined by the remote web server. How does it know whether the client is python 2 or python 3?

Reply:
Has been updated. Now the gzipped body will be unzipped in one place.